implants/lib/eldritch/src/pivot/arp_scan_impl.rs

```diff
@@ -1,5 +1,296 @@
+fn start_listener(
+    interface: NetworkInterface,
+    data: Arc<Mutex<HashMap<Ipv4Addr, Option<ArpResponse>>>>,
+) {
+    if interface.ips.iter().filter(|ip| ip.is_ipv4()).count() == 0 {
+        return;
+    }
+    let mac = match interface.mac {
+        Some(mac) => mac,
+        None => return,
+    };
+    let (mut tx, mut rx) = match channel(&interface, Default::default()) {
+        Ok(Ethernet(tx, rx)) => (tx, rx),
+        Ok(_) => panic!("Unhandled channel type"),
+        Err(e) => {
+            println!("Error creating channel: {}", e);
+            return;
+        }
+    };
+    let ips = match data.lock() {
+        Ok(lock) => lock.keys().cloned().collect::<Vec<Ipv4Addr>>(),
+        Err(err) => {
+            println!("Failed to get lock on ips: {}", err);
+            return;
+        }
+    };
+    for ip in ips {
+        let ip_to_use = match interface
+            .ips
+            .iter()
+            .find(|interface_ip| interface_ip.contains(IpAddr::V4(ip)))
+        {
+            Some(IpNetwork::V4(ipnet)) => ipnet,
+            Some(_) => continue,
+            None => continue,
+        };
+        let mut arp_packet = [0u8; 28];
+        let mut arp = match MutableArpPacket::new(&mut arp_packet) {
+            Some(arp) => arp,
+            None => {
+                println!("Failed to create MutableArpPacket to send.");
+                return;
+            }
+        };
+        arp.set_hardware_type(pnet::packet::arp::ArpHardwareType(1));
+        arp.set_protocol_type(pnet::packet::ethernet::EtherType(0x0800));
+        arp.set_hw_addr_len(6);
+        arp.set_proto_addr_len(4);
+        arp.set_operation(ArpOperations::Request);
+        arp.set_sender_hw_addr(mac);
+        arp.set_sender_proto_addr(ip_to_use.ip());
+        arp.set_target_hw_addr(MacAddr::zero());
+        arp.set_target_proto_addr(ip);
+        let mut eth_packet = [0u8; 60];
+        let mut eth = match MutableEthernetPacket::new(&mut eth_packet) {
+            Some(eth) => eth,
+            None => {
+                println!("Failed to create MutableEthernetPacket to send.");
+                return;
+            }
+        };
+        eth.set_destination(MacAddr::broadcast());
+        eth.set_source(mac);
+        eth.set_ethertype(EtherType(0x0806));
+        eth.set_payload(arp.packet());
+        match tx.send_to(eth.packet(), None) {
+            Some(Ok(_)) => {}
+            Some(Err(err)) => {
+                println!("Failed to tx on {}: {}", interface.name, err);
+                return;
+            }
+            None => {
+                println!("Failed to tx on {}: Returned None", interface.name);
+                return;
+            }
+        };
+        let now = SystemTime::now();
+        loop {
+            let elapsed = match now.elapsed() {
+                Ok(elapsed) => elapsed,
+                Err(err) => {
+                    println!("Failed to get elapsed time on {}: {}", interface.name, err);
```

Comment:
We want to avoid println in production code eldritch should only print when a user calls print (print calls in eldritch have a special handler to ensure they get back to the c2 rust prints wont get recorded). If you want to preserve debug strings like this you can wrap them in an if debug and set debug to false.